pkg/resources/cloudcontroller/deployment.go

```diff
@@ -39,67 +39,69 @@ const (
 func DeploymentReconciler(data *resources.TemplateData) reconciling.NamedDeploymentReconcilerFactory {
 	var creatorGetter reconciling.NamedDeploymentReconcilerFactory
 
-	switch {
-	case data.Cluster().Spec.Cloud.AWS != nil:
-		creatorGetter = awsDeploymentReconciler(data)
+	if data.Cluster().Spec.HasCCM() {
```

Comment:
I don't get this. What exactly is this extra condition trying to accomplish?

Reply:
> In addition, I'm wrapping the cloud controller deployment reconciler in the new HasCCM method to make sure that people implementing new CCMs see this method and make sure it's updated.

Reply:
If you add a CCM and forget to update HasCCM, surely it won't work because some webhook will then refuse to enable the CCM. So you'd notice that you have to extend HasCCM as well.

If you only extend HasCCM, you will notice that nothing gets deployed and that you have to extend this function.

This feels like a runtime check for something that shouldn't exist or at the worst should be a unit test to prevent drift between the two functions.